src/frontend/js/modules/i18n/i18n-string.service.js

```diff
@@ -1,25 +1,29 @@
-(function(angular) {
-  'use strict';
+'use strict';
 
-  angular.module('esn.i18n')
-    .factory('EsnI18nString', EsnI18nStringFactory);
+angular.module('esn.i18n')
+  .factory('EsnI18nString', EsnI18nStringFactory);
 
-    function EsnI18nStringFactory($translate) {
-      function EsnI18nString(text, params) {
-        this.text = text;
-        this.params = params;
+  function EsnI18nStringFactory($translate) {
+    // TODO: Write tests for this (https://github.com/OpenPaaS-Suite/esn-frontend-common-libs/issues/83)
+    function EsnI18nString(text, interpolateParams = {}, ignoreSanitizeStrategy = false) {
+      this.text = text;
+      this.interpolateParams = interpolateParams;
+      this.ignoreSanitizeStrategy = ignoreSanitizeStrategy;
 
-        return this;
-      }
+      return this;
+    }
 
-      EsnI18nString.prototype.toString = function() {
-        if (!this.translated) {
-          this.translated = $translate.instant(this.text, this.params);
+    EsnI18nString.prototype.toString = function() {
+      if (!this.translated) {
+        if (this.ignoreSanitizeStrategy) {
+          this.translated = $translate.instant(this.text, this.interpolateParams, undefined, undefined, null);
+        } else {
+          this.translated = $translate.instant(this.text, this.interpolateParams);
```

Comment:
```suggestion
      if (this.translated) return this.translated;

      return this.ignoreSanitizeStrategy ? $translate.instant(this.text, this.interpolateParams, undefined, undefined, null) : $translate.instant(this.text, this.interpolateParams);
```

Reply:
Your solution is lack of step to assign the translated value to this.translated